content/docs/testing-recipes.md

```diff
@@ -95,46 +78,42 @@ export default function User(props) {
 We can write tests for it:
 
 ```jsx
+// user.test.js
+import { render } from "@testing-library/react";
 import User from "./user";
-import { render } from "react-dom";
-import { act } from "react-dom/test-utils";
 
-let container = null
 beforeEach(() => {
-  // setup a DOM element as a render target
-  container = document.createElement('div')
-  document.body.appendChild(container)
-})
+  // mock window.fetch so the unit tests don't actually make HTTP requests
+  jest.spyOn(window, "fetch").mockImplementation(() => {});
+});
 
 afterEach(() => {
-  // cleanup on exiting
-  container.remove();
-  container = null  
-})
+  // remove the mock to ensure tests are completely isolated
+  window.fetch.mockRestore();
+});
 
-it("renders user data", async () => {
+test("renders user data", async () => {
   const fakeUser = {
     name: "Joni Baez",
     age: "32",
-    address: "123, Charming Avenue"
+    address: "123, Charming Avenue",
   };
-  jest.spyOn(global, "fetch").mockImplementation(() =>
+  window.fetch.mockImplementationOnce(() =>
```

Comment:
When it comes to mocks, I usually set `resetMocks` or `clearMocks` globally in Jest. There is rarely a reason to maintain mock state across tests.

https://jestjs.io/docs/en/configuration#resetmocks-boolean

Reply:
Maybe we should add a comment about that so people can look into it if they want to reduce the boilerplate